### DISCUSSION TITLE
[FR] Country field

### DISCUSSION THREAD

**Discussion by jmauzyk** [craftcms/cms#13789] [Ideas] [answered]
Now that addresses are elements in Craft, I'm wondering if it would be possible to have a country field or be able to auto-generate country options for a dropdown field. It would be nice to be able to leverage the existing country functionality rather than having to manually create all the country options.

**Comment by scandella** [comment]
[This plugin might help.](https://plugins.craftcms.com/craft-dynamic-fields?craft4)

**Reply by brandonkelly** [reply]
Yep that would work in the meantime. Your options template could return:

```twig
{{ craft.app.addresses.countryRepository.getList(craft.app.language)|json_encode|raw }}
```

**Comment by brandonkelly** [comment, marked as answer]
Good idea. Just added a Country field type for Craft 4.6 and 5.0 (6f3e2e7def4980e3aedc79b5a9f61869511564ee).

**Reply by brandonkelly** [reply]
Craft 4.6.0 is out now with the new Country field type.
